viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    return friends_movie_list
+
+def users_list(user_data):
+    users_movie_list= []
+    for movie in user_data["watched"]:
+        if movie not in users_movie_list:
+            users_movie_list.append(movie)
+    return users_movie_list
+
+def get_unique_watched(user_data):
+    user_unique_list=[]
+    friends_movies = friends_list(user_data)
+    users_movies = users_list(user_data)
+   
+    for movie in users_movies:
+        if movie not in friends_movies:
+            user_unique_list.append(movie)
+    return user_unique_list
+    # print(unique_list == friends_movies - users_movies)
+    # return unique_list == friends_movies - users_movies 
+    # user_unique_list == list(set(friends_movies)-set(users_movies))     
+    # return user_unique_list
+
+def get_friends_unique_watched(user_data):
+    friend_unique_list=[]
+    friends_movies = friends_list(user_data)
+    users_movies = users_list(user_data)
+
+    for movie in friends_movies:
+        if movie not in users_movies:
+            friend_unique_list.append(movie)
+    return friend_unique_list
+#     # print(unique_list == friends_movies - users_movies)
+#     # return unique_list == friends_movies - users_movies 
+#     friend_unique_list == list(set(friends_movies)-set(users_movies))     
+#     return friend_unique_list
+
+    # for movie in user_data["watched"]:
+    #     for friend in user_data["friends"]:
+    #         for friend_movie in friend["watched"]:
+    #             if movie["title"] ==
+    #         # user_data["watched"] not in friends["watched"]:
+    # unique_list.append(value)
+     
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    rec_movies = []
+    friend_unique = get_friends_unique_watched(user_data)
+    for movie in friend_unique:
+        if movie["host"]==[] or user_data["subscriptions"]==[]:
+            return 0
+        if movie["host"] in user_data["subscriptions"]:
+            rec_movies.append(movie)
+    return rec_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    genre_rec = []
+    genre = get_most_watched_genre(user_data)
+    friend_unique = get_friends_unique_watched(user_data)
+    if not user_data["watched"] or not friend_unique:
+        return genre_rec
+    for movie in friend_unique:
+        if movie["genre"] in genre:
+            genre_rec.append(movie)
+    return genre_rec
 
+def get_rec_from_favorites(user_data):
+    rec_movies = []
+    user_unique = get_unique_watched(user_data)
+    if not user_data["favorites"] or not user_unique:
+        return 0

```

Comment:
Same comment as above, I would suggest you return an empty list so that your function is only returning one data type -- list.